File: frontend/src/lib/api/metadata.ts
export interface Department {
  code: string;
  name: string;
  sections: string[];
}

export interface DocumentType {
  code: string;
  description: string;
  root_directory: string | null;
  active: boolean;
}

export async function fetchDepartments(): Promise<Department[]> {
  const res = await fetch('/api/departments');
  if (!res.ok) throw new Error('Failed to fetch departments');
  return res.json();
}

export async function fetchDocumentTypes(): Promise<DocumentType[]> {
  const res = await fetch('/api/document-types');
  if (!res.ok) throw new Error('Failed to fetch document types');
  return res.json();
}

File: backend/src/api/metadata.rs
//! Metadata API handlers (departments, document types)

use axum::{extract::State, Json};
use sqlx::SqlitePool;

use crate::error::Result;
use crate::storage::{department, document_type};

/// GET /api/departments
pub async fn list_departments(State(pool): State<SqlitePool>) -> Result<Json<serde_json::Value>> {
    let deps = department::list_departments(&pool).await?;
    Ok(Json(serde_json::to_value(deps)?))
}

/// GET /api/document-types
pub async fn list_document_types(State(pool): State<SqlitePool>) -> Result<Json<serde_json::Value>> {
    let types = document_type::list_active_document_types(&pool).await?;
    Ok(Json(serde_json::to_value(types)?))
}
